src/functional/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -16,24 +16,28 @@
 import dataclasses
 from typing import Any, Final, TypeVar
 
-import numpy as np
-
+import functional.otf.binding.type_specifications as ts
+from functional.iterator.embedded import IndexField
 from functional.otf import languages, stages, step_types, workflow
 from functional.otf.binding import cpp_interface, interface
 from functional.program_processors.codegens.gtfn import gtfn_backend
+from functional.type_system import type_translation
 
 
 T = TypeVar("T")
 
 
-def get_param_description(
-    name: str, obj: Any
-) -> interface.ScalarParameter | interface.BufferParameter:
-    view: np.ndarray = np.asarray(obj)
-    if view.ndim > 0:
-        return interface.BufferParameter(name, tuple(dim.value for dim in obj.axes), view.dtype)
+def get_param_description(name: str, obj: Any) -> interface.Parameter:
+    # TODO: make type_translation extendable and specialize it for the
+    #   bindings to distinguish between buffer-backed fields and index
+    #   fields.
```

Comment:
I now understand that the comment about _making `type_translation` extendable_ is related to use a different mechanism for the pattern matching part with the actual type, and not with the preliminary handling of the messy implementation details of python typing annotations. It makes sense, but still don't see it as a high priority for now.

Reply:
I wrote a new `from_value` function that tries to match for `IndexFieldType` and if it fails it delegates it to the original `functional.type_system.from_value` function.

Reply:
I'd like to use a registry that allows you to register a match functor for every type of the `type_system`, then you can extend without raising the cyclomatic complexity through all these if statements and you can also simply chain the list of matchers to combine modules.

Reply:
That would also replace the `__gt_type__` solution which is an intrusive implementation because it forces objects to be aware of the type system.